keel-core/src/main/kotlin/com/netflix/spinnaker/keel/persistence/memory/InMemoryResourceRepository.kt

```diff
@@ -120,13 +120,19 @@ class InMemoryResourceRepository(
     appendHistory(applicationEvents, event)
   }
 
-  private fun <T : PersistentEvent> appendHistory(eventList: MutableMap<String, MutableList<T>>, event: T) {
+  private fun <T : PersistentEvent> appendHistory(
+    eventList: MutableMap<String, MutableList<T>>,
+    event: T
+  ) {
     eventList.computeIfAbsent(event.uid) {
       mutableListOf()
     }
       .let {
         val mostRecentEvent = it.firstOrNull() // we get the first because the list is in descending order
-        if (!event.ignoreRepeatedInHistory || event.javaClass != mostRecentEvent?.javaClass) {
+        if (event.javaClass == mostRecentEvent?.javaClass && event.ignoreRepeatedInHistory) {
+          it.removeAt(0)
```

Comment:
What is happening here with the `add`/`remove`? it feels a bit dangerous.. can you add some comments?

Reply:
Unfortunately there’s no `replace` method on `List`, and I can’t change the existing event because it’s a data class. But this is effective just replacing the event with another one of the same type but with a newer timestamp. And this class is going away next week after @robfletcher’s PR. 😊